### PR TITLE
feat: adding metric_url for reth

### PR DESCRIPTION
**Pull request by lucaprov1989**
Added metric_url in el context client

Linked to: https://github.com/kurtosis-tech/eth2-package/pull/195

### CODE REVIEW

File: src/el/el_client_context.star [hunk 1]
```diff
@@ -7,6 +7,7 @@ def new_el_client_context(
 	ws_port_num,
 	engine_rpc_port_num,
 	jwt_secret,
+	metric_url,
```
Comment (h4ck3rk3y) [inline src/el/el_client_context.star:10]: can we make this the last thing we pass and put `""` at the end, this ensures backwards compatibilty with any other callers of this function that haven't started passing metric_url,

File: src/el/besu/besu_launcher.star [hunk 2]
```diff
@@ -97,6 +97,7 @@ def launch(
 		WS_PORT_NUM,
 		ENGINE_HTTP_RPC_PORT_NUM,
 		jwt_secret,
+		"", # Passing empty metric_url for now
```
Comment (h4ck3rk3y) [inline src/el/besu/besu_launcher.star:100]: Can we stick in `TODO` here

File: src/el/erigon/erigon_launcher.star [hunk 3]
```diff
@@ -94,6 +94,7 @@ def launch(
 		WS_RPC_PORT_NUM,
 		ENGINE_RPC_PORT_NUM,
 		jwt_secret,
+		"", # Passing empty metric_url for now
```
Comment (h4ck3rk3y) [inline src/el/erigon/erigon_launcher.star:97]: lets make this `TODO`

File: src/el/ethereumjs/ethereumjs_launcher.star [hunk 4]
```diff
@@ -109,6 +109,7 @@ def launch(
 		WS_PORT_NUM,
 		ENGINE_RPC_PORT_NUM,
 		jwt_secret,
+		"", # Passing empty metric_url for now
```
Comment (h4ck3rk3y) [inline src/el/ethereumjs/ethereumjs_launcher.star:112]: lets stick in a `TODO`

File: src/el/reth/reth_launcher.star [hunk 5]
```diff
@@ -97,6 +97,8 @@ def launch(
 
 	jwt_secret = shared_utils.read_file_from_service(plan, service_name, jwt_secret_json_filepath_on_client)
 
+	metric_url = "http://{0}:{1}".format(service.ip_address, METRICS_PORT_NUM)
```
Comment (h4ck3rk3y) [inline src/el/reth/reth_launcher.star:100]: this looks good to me!

**Comment by cbermudez97** [comment]
@lucaprov1989 @h4ck3rk3y I found having https://github.com/kurtosis-tech/eth2-package/pull/195 really useful and found out this PR adding some of the requirements for it. Created https://github.com/lucaprov1989/eth-network-package/pull/1 to add configurations for other EL besides Reth. I would appreciate it if we could include those changes here too. 👍

**Comment by lucaprov1989** [comment]
> @lucaprov1989 @h4ck3rk3y I found having [kurtosis-tech/eth2-package#195](https://github.com/kurtosis-tech/eth2-package/pull/195) really useful and found out this PR adding some of the requirements for it. Created [lucaprov1989#1](https://github.com/lucaprov1989/eth-network-package/pull/1) to add configurations for other EL besides Reth. I would appreciate it if we could include those changes here too. 👍

Hey @cbermudez97 thanks for this! I have just merged your integration.

File: src/el/besu/besu_launcher.star [hunk 6]
```diff
@@ -97,6 +106,7 @@ def launch(
 		WS_PORT_NUM,
 		ENGINE_HTTP_RPC_PORT_NUM,
 		jwt_secret,
+		besu_metrics_info,
```
Comment (h4ck3rk3y) [inline src/el/besu/besu_launcher.star:109]: can the metrics info be the absolute last thing? helps with backwards compatibility

File: src/el/el_client_context.star [hunk 7]
```diff
@@ -7,6 +7,7 @@ def new_el_client_context(
 	ws_port_num,
 	engine_rpc_port_num,
 	jwt_secret,
+	el_metrics_info,
```
Comment (h4ck3rk3y) [inline src/el/el_client_context.star:10]: can this be the absolute last thing and default it to None `el_metrics_info=None`

File: src/el/ethereumjs/ethereumjs_launcher.star [hunk 8]
```diff
@@ -44,7 +47,7 @@ USED_PORTS = {
 	TCP_DISCOVERY_PORT_ID: shared_utils.new_port_spec(DISCOVERY_PORT_NUM, shared_utils.TCP_PROTOCOL),
 	UDP_DISCOVERY_PORT_ID: shared_utils.new_port_spec(DISCOVERY_PORT_NUM, shared_utils.UDP_PROTOCOL),
 	ENGINE_RPC_PORT_ID: shared_utils.new_port_spec(ENGINE_RPC_PORT_NUM, shared_utils.TCP_PROTOCOL),
-#    METRICS_PORT_ID: shared_utils.new_port_spec(METRICS_PORT_NUM, shared_utils.TCP_PROTOCOL)
+    # METRICS_PORT_ID: shared_utils.new_port_spec(METRICS_PORT_NUM, shared_utils.TCP_PROTOCOL)
```
Comment (h4ck3rk3y) [inline src/el/ethereumjs/ethereumjs_launcher.star:50]: do you want this to be commented? is this a bug?

Comment (h4ck3rk3y) [reply src/el/ethereumjs/ethereumjs_launcher.star:50]: oh you are figuring it out i suppose

File: src/el/ethereumjs/ethereumjs_launcher.star [hunk 9]
```diff
@@ -100,6 +103,10 @@ def launch(
 
 	jwt_secret = shared_utils.read_file_from_service(plan, service_name, jwt_secret_json_filepath_on_client)
 
+	# TODO: Passing empty string for metrics_url for now
+	# metrics_url = "http://{0}:{1}".format(service.ip_address, METRICS_PORT_NUM)
+	ethjs_metrics_info = node_metrics.new_node_metrics_info(service_name, METRICS_PATH, "")
```
Comment (h4ck3rk3y) [inline src/el/ethereumjs/ethereumjs_launcher.star:108]: lets add an issue to the package tracking and put a comment here to it!

**Review by h4ck3rk3y — APPROVED** [review]
LGTM, just a few minor changes left

File: src/el/ethereumjs/ethereumjs_launcher.star (hunk 9 above)
Comment (cbermudez97) [reply src/el/ethereumjs/ethereumjs_launcher.star:108]: Created #127 for this. Will add it to the comment.

**Pull request merged**